src/agent/src/sandbox.rs

```diff
@@ -261,6 +265,39 @@ impl Sandbox {
 
         Ok(())
     }
+
+    pub fn add_hooks(&mut self, dir: &str) -> Result<()> {
+        let mut hooks = Hooks::default();
+        hooks.prestart = self.find_hooks(dir, "prestart")?;
```

Comment:
Will all of  those three hooks exist? For example if poststart hook doesn't exist, then it will return directly an error info instead of skip it.

Reply:
Good point! Thanks!